homeassistant/components/garages_amsterdam/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The Garages Amsterdam integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+import garages_amsterdam
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, update_coordinator
+
+from .const import DOMAIN
+
+PLATFORMS = ["binary_sensor", "sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Garages Amsterdam component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Garages Amsterdam from a config entry."""
+    await get_coordinator(hass)
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+
+    return unload_ok
```

Comment:
We need to clean up hass.data of the coordinator if it's the last config entry before we return here.

Use `hass.config_entries.async_entries` to get a list of all entries of the domain.

Reply:
```suggestion
    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
        hass.data.pop(DOMAIN)

    return unload_ok
```